File: frontend/olive-platform/src/features/plots/data/responses/PlotDetailsResponse.ts
import type { PlotVarietyDetailResponse } from "./PlotVarietyDetailResponse";

export interface PlotDetailsResponse {
  id: number;
  reference: string;
  name: string;
  areaHectares: number | null;
  numberOfTrees: number;
  plantingYear: number | null;
  location: string | null;
  notes: string | null;
  createdAt: string;
  harvestedTreesPercentage: number;
  plannedTreesPercentage: number;
  canLaunchHarvest: boolean;
  varieties: PlotVarietyDetailResponse[]
}

File: backend/OlivePlatform/OlivePlatform.Application/Features/Plots/Responses/PlotDetailsResponse.cs
namespace OlivePlatform.Application.Features.Plots.Responses;

public class PlotDetailResponse
{
    public int Id { get; set; }
    public string Reference { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public decimal? AreaHectares { get; set; }
    public int NumberOfTrees { get; set; }
    public int? PlantingYear { get; set; }
    public string? Location { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }

    public double HarvestedTreesPercentage { get; set; }

    public double PlannedTreesPercentage { get; set; }
    public bool CanLaunchHarvest { get; set; }

    public List<PlotVarietyDetail> Varieties { get; set; } = new();
}